core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1167,9 +1274,8 @@ private void processParse(
 
             LOG.debug().$("params [count=").$(parameterCount).$(']').$();
             lo += Short.BYTES;
-
             bindVariableService.clear();
-            setupBindVariables(lo, parameterCount, bindVariableSetters);
+            setupBindVariables(lo, parameterCount, bindVariableSetters, bindVariableTypes);
```

Comment:
`bindVariableTypes` should perhaps be a simple java object to begin with. I would add a simple to-do item that we want to re-use these lists.

Realistically reuse is only possible when "statement name" disappears from cache. Then we can add IntList back to some sort of pool.

Reply:
we can create these IntLists lazily only when parameterCount > 0